src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -514,8 +514,58 @@ void bulk_load_service::handle_app_downloading(const bulk_load_response &respons
 void bulk_load_service::handle_app_ingestion(const bulk_load_response &response,
                                              const rpc_address &primary_addr)
 {
-    // TODO(heyuchen): TBD
-    // called by `on_partition_bulk_load_reply` when app status is ingesting
+    const std::string &app_name = response.app_name;
+    const gpid &pid = response.pid;
+
+    if (!response.__isset.is_group_ingestion_finished) {
+        dwarn_f("recevie bulk load response from node({}) app({}) partition({}), "
+                "primary_status({}), but is_group_ingestion_finished is not set",
+                primary_addr.to_string(),
+                response.app_name,
+                pid,
+                dsn::enum_to_string(response.primary_bulk_load_status));
+        return;
+    }
+
+    for (const auto &kv : response.group_bulk_load_state) {
+        const auto &bulk_load_states = kv.second;
+        if (!bulk_load_states.__isset.ingest_status) {
+            dwarn_f("recevie bulk load response from node({}) app({}) partition({}), "
+                    "primary_status({}), but node({}) ingestion_status is not set",
+                    primary_addr.to_string(),
+                    response.app_name,
+                    pid,
+                    dsn::enum_to_string(response.primary_bulk_load_status),
+                    kv.first.to_string());
+            return;
+        }
+
+        if (bulk_load_states.ingest_status == ingestion_status::IS_FAILED) {
+            derror_f("app({}) partition({}) node({}) ingestion failed",
```

Comment:
Keep the same order as L533?

Reply:
This log format is like L474-L479, shows which node ingestion failed.